### PR TITLE
Refactor child process management into library

### LINKED ISSUE #3128: Shelling out to kill is fragile and should be avoided
The crash message:

```  ☠  Coda Daemon crashed. The Coda Protocol developers would like to know why!

  Please:
    Open an issue:
      https://github.com/CodaProtocol/coda/issues/new

    Briefly describe what you were doing, and include the last 20 lines from .coda-config/coda.log.
    And then paste the following:

    (monitor.ml.Error
  (Unix.Unix_error "No such file or directory" Core.Unix.create_process
    "((prog kill) (args (124)) (env (Extend ())))")
  ("Raised at file \"src/error.ml\" (inlined), line 9, characters 14-30"
    "Called from file \"src/or_error.ml\", line 72, characters 17-32"
    "Called from file \"src/deferred1.ml\", line 17, characters 40-45"
    "Called from file \"src/job_queue.ml\" (inlined), line 131, characters 2-5"
    "Called from file \"src/job_queue.ml\", line 170, characters 6-47"
    "Caught by monitor coda"))
```

coda.log content:

```
{"timestamp":"2019-08-07 20:29:26.184840Z","level":"Trace","source":{"module":"Gossip_net","location":"File \"lib/gossip_net/gossip_net.ml\", line 473, characters 71-78"},"message":"broadcasting message","metadata":{"pid":7}}
{"timestamp":"2019-08-07 20:29:26.221708Z","level":"Debug","source":{"module":"Trust_system__Peer_trust","location":"File \"lib/trust_system/peer_trust.ml\", line 120, characters 56-63"},"message":"Decreasing trust for peer $peer due to action Trust_system.Actions.Sent_old_gossip. New trust is 0.000006.","metadata":{"peer":"52.53.238.133","pid":7}}
{"timestamp":"2019-08-07 20:29:26.221940Z","level":"Debug","source":{"module":"Network_pool__Network_pool_base","location":"File \"lib/network_pool/network_pool_base.ml\", line 32, characters 60-67"},"message":"Pool diff apply feedback: no useful transactions","metadata":{"pid":7}}
{"timestamp":"2019-08-07 20:29:26.235287Z","level":"Debug","source":{"module":"Trust_system__Peer_trust","location":"File \"lib/trust_system/peer_trust.ml\", line 120, characters 56-63"},"message":"Decreasing trust for peer $peer due to action Trust_system.Actions.Sent_old_gossip. New trust is 0.000021.","metadata":{"peer":"52.33.52.99","pid":7}}
{"timestamp":"2019-08-07 20:29:26.235602Z","level":"Debug","source":{"module":"Network_pool__Network_pool_base","location":"File \"lib/network_pool/network_pool_base.ml\", line 32, characters 60-67"},"message":"Pool diff apply feedback: no useful transactions","metadata":{"pid":7}}
{"timestamp":"2019-08-07 20:29:26.249760Z","level":"Debug","source":{"module":"Trust_system__Peer_trust","location":"File \"lib/trust_system/peer_trust.ml\", line 120, characters 56-63"},"message":"Decreasing trust for peer $peer due to action Trust_system.Actions.Sent_old_gossip. New trust is 0.000033.","metadata":{"peer":"54.188.108.191","pid":7}}
{"timestamp":"2019-08-07 20:29:26.249930Z","level":"Debug","source":{"module":"Network_pool__Network_pool_base","location":"File \"lib/network_pool/network_pool_base.ml\", line 32, characters 60-67"},"message":"Pool diff apply feedback: no useful transactions","metadata":{"pid":7}}
{"timestamp":"2019-08-07 20:29:26.268336Z","level":"Debug","source":{"module":"Trust_system__Peer_trust","location":"File \"lib/trust_system/peer_trust.ml\", line 120, characters 56-63"},"message":"Decreasing trust for peer $peer due to action Trust_system.Actions.Sent_old_gossip. New trust is 0.000008.","metadata":{"peer":"54.184.87.39","pid":7}}
{"timestamp":"2019-08-07 20:29:26.268503Z","level":"Debug","source":{"module":"Network_pool__Network_pool_base","location":"File \"lib/network_pool/network_pool_base.ml\", line 32, characters 60-67"},"message":"Pool diff apply feedback: no useful transactions","metadata":{"pid":7}}
{"timestamp":"2019-08-07 20:29:26.285923Z","level":"Debug","source":{"module":"Trust_system__Peer_trust","location":"File \"lib/trust_system/peer_trust.ml\", line 120, characters 56-63"},"message":"Decreasing trust for peer $peer due to action Trust_system.Actions.Sent_old_gossip. New trust is 0.000049.","metadata":{"peer":"54.149.124.61","pid":7}}
{"timestamp":"2019-08-07 20:29:26.286231Z","level":"Debug","source":{"module":"Network_pool__Network_pool_base","location":"File \"lib/network_pool/network_pool_base.ml\", line 32, characters 60-67"},"message":"Pool diff apply feedback: no useful transactions","metadata":{"pid":7}}
{"timestamp":"2019-08-07 20:29:26.385732Z","level":"Debug","source":{"module":"Trust_system__Peer_trust","location":"File \"lib/trust_system/peer_trust.ml\", line 120, characters 56-63"},"message":"Decreasing trust for peer $peer due to action Trust_system.Actions.Sent_old_gossip. New trust is 0.000019.","metadata":{"peer":"54.193.108.137","pid":7}}
{"timestamp":"2019-08-07 20:29:26.385865Z","level":"Debug","source":{"module":"Network_pool__Network_pool_base","location":"File \"lib/network_pool/network_pool_base.ml\", line 32, characters 60-67"},"message":"Pool diff apply feedback: no useful transactions","metadata":{"pid":7}}
{"timestamp":"2019-08-07 20:29:27.895808Z","level":"Debug","source":{"module":"Trust_system__Peer_trust","location":"File \"lib/trust_system/peer_trust.ml\", line 120, characters 56-63"},"message":"Decreasing trust for peer $peer due to action Trust_system.Actions.Connected. New trust is 0.000008.","metadata":{"peer":"54.184.87.39","pid":7}}
{"timestamp":"2019-08-07 20:29:27.963310Z","level":"Debug","source":{"module":"Trust_system__Peer_trust","location":"File \"lib/trust_system/peer_trust.ml\", line 120, characters 56-63"},"message":"Decreasing trust for peer $peer due to action Trust_system.Actions.Connected. New trust is 0.000032.","metadata":{"peer":"54.188.108.191","pid":7}}
{"timestamp":"2019-08-07 20:29:27.973336Z","level":"Debug","source":{"module":"Trust_system__Peer_trust","location":"File \"lib/trust_system/peer_trust.ml\", line 120, characters 56-63"},"message":"Decreasing trust for peer $peer due to action Trust_system.Actions.Connected. New trust is 0.000006.","metadata":{"peer":"52.53.238.133","pid":7}}
{"timestamp":"2019-08-07 20:29:28.242982Z","level":"Debug","source":{"module":"Trust_system__Peer_trust","location":"File \"lib/trust_system/peer_trust.ml\", line 120, characters 56-63"},"message":"Increasing trust for peer $peer due to action Trust_system.Actions.Sent_useful_gossip ($cmd). New trust is 0.000012.","metadata":{"cmd":{"payload":{"common":{"fee":5,"nonce":"3","memo":"2pmu64f2x97tNiDXMycnLwBSECDKbX77MTXVWVsG8hcRFsedhXDWWq"},"body":["Payment",{"receiver":"CfJpEumzrhtqsLBw2hJeE8sc9Tpeii5xAynrtvYeuYNqeU7waThkFnijJa5v5kWf1w7KS7FNkXA1hyPXRZQCnUe7Xuk8hH54esMDXhyEiGHuxEWvbgLnXJTnJkM7xg5PbyQXso9xk6TyJ","amount":20}]},"sender":"JzXj6kK5m3HYvtvecLFt26mkMqFxSvvt9uJMwVjPVmZ76aVo6Qh3W1ELw8Eid63pYGiVU2yEc18LCe6r9akvRx72pqAVFzU2fLPHRDnSw1R7ysexL7BfRxk9rLQkhdAkQPFvhAdq3Sc3wrryQEU9JZ9gLrAtAgGwXVCdM7xb7cgpCFQoAP4BCYtjAC2fNUQf5n3eYkuVmfQsFPVur7Ani3j4KTADdrcz135kYWMTtsEE5sRw5mhepmLuF9kRJHktDTMNwaPvVarnZaGE","signature":"EXQ8XhSnNt542kR7jvDgWbs3veW2stnTZqUMPNVoiqoiztGrR6km6bz2tiMDpiHLMcnQVQmmToghAXH4fads1eZ1UScftDs6LFrj4onLK5ezxdhi2Xqtruc7udbYzR7TFi3iYVqo6DUfR7FFkLxFQ4UwaZBXfjppRSWfDVPiSiV7uJVr6MEkvsiztLDhPT8udeDtp1L2m29JyhHZRaVmRSwfZTcJQ9X6BxBUKTnMTb2JLAaPzupoAoggXUVt2ZXw7wpZaicDtQZfeH12"},"peer":"54.184.87.39","pid":7}}
{"timestamp":"2019-08-07 20:29:28.243174Z","level":"Trace","source":{"module":"Network_pool__Network_pool_base","location":"File \"lib/network_pool/network_pool_base.ml\", line 27, characters 60-67"},"message":"Broadcasting Transaction diff of length 1","metadata":{"pid":7}}
{"timestamp":"2019-08-07 20:29:28.247674Z","level":"Trace","source":{"module":"Coda_networking","location":"File \"lib/coda_networking/coda_networking.ml\", line 649, characters 56-63"},"message":"Broadcasting transaction pool diff over gossip net","metadata":{"message":["Transaction_pool_diff",[{"payload":{"common":{"fee":5,"nonce":"3","memo":"2pmu64f2x97tNiDXMycnLwBSECDKbX77MTXVWVsG8hcRFsedhXDWWq"},"body":["Payment",{"receiver":"CfJpEumzrhtqsLBw2hJeE8sc9Tpeii5xAynrtvYeuYNqeU7waThkFnijJa5v5kWf1w7KS7FNkXA1hyPXRZQCnUe7Xuk8hH54esMDXhyEiGHuxEWvbgLnXJTnJkM7xg5PbyQXso9xk6TyJ","amount":20}]},"sender":"JzXj6kK5m3HYvtvecLFt26mkMqFxSvvt9uJMwVjPVmZ76aVo6Qh3W1ELw8Eid63pYGiVU2yEc18LCe6r9akvRx72pqAVFzU2fLPHRDnSw1R7ysexL7BfRxk9rLQkhdAkQPFvhAdq3Sc3wrryQEU9JZ9gLrAtAgGwXVCdM7xb7cgpCFQoAP4BCYtjAC2fNUQf5n3eYkuVmfQsFPVur7Ani3j4KTADdrcz135kYWMTtsEE5sRw5mhepmLuF9kRJHktDTMNwaPvVarnZaGE","signature":"EXQ8XhSnNt542kR7jvDgWbs3veW2stnTZqUMPNVoiqoiztGrR6km6bz2tiMDpiHLMcnQVQmmToghAXH4fads1eZ1UScftDs6LFrj4onLK5ezxdhi2Xqtruc7udbYzR7TFi3iYVqo6DUfR7FFkLxFQ4UwaZBXfjppRSWfDVPiSiV7uJVr6MEkvsiztLDhPT8udeDtp1L2m29JyhHZRaVmRSwfZTcJQ9X6BxBUKTnMTb2JLAaPzupoAoggXUVt2ZXw7wpZaicDtQZfeH12"}]],"pid":7}}
{"timestamp":"2019-08-07 20:29:28.247776Z","level":"Trace","source":{"module":"Gossip_net","location":"File \"lib/gossip_net/gossip_net.ml\", line 473, characters 71-78"},"message":"broadcasting message","metadata":{"pid":7}}
{"timestamp":"2019-08-07 20:29:28.288623Z","level":"Debug","source":{"module":"Trust_system__Peer_trust","location":"File \"lib/trust_system/peer_trust.ml\", line 120, characters 56-63"},"message":"Decreasing trust for peer $peer due to action Trust_system.Actions.Sent_old_gossip. New trust is 0.000006.","metadata":{"peer":"52.53.238.133","pid":7}}
{"timestamp":"2019-08-07 20:29:28.288759Z","level":"Debug","source":{"module":"Network_pool__Network_pool_base","location":"File \"lib/network_pool/network_pool_base.ml\", line 32, characters 60-67"},"message":"Pool diff apply feedback: no useful transactions","metadata":{"pid":7}}
{"timestamp":"2019-08-07 20:29:28.317957Z","level":"Debug","source":{"module":"Trust_system__Peer_trust","location":"File \"lib/trust_system/peer_trust.ml\", line 120, characters 56-63"},"message":"Decreasing trust for peer $peer due to action Trust_system.Actions.Sent_old_gossip. New trust is 0.000032.","metadata":{"peer":"54.188.108.191","pid":7}}
{"timestamp":"2019-08-07 20:29:28.318153Z","level":"Debug","source":{"module":"Network_pool__Network_pool_base","location":"File \"lib/network_pool/network_pool_base.ml\", line 32, characters 60-67"},"message":"Pool diff apply feedback: no useful transactions","metadata":{"pid":7}}
{"timestamp":"2019-08-07 20:29:28.518308Z","level":"Debug","source":{"module":"Trust_system__Peer_trust","location":"File \"lib/trust_system/peer_trust.ml\", line 120, characters 56-63"},"message":"Decreasing trust for peer $peer due to action Trust_system.Actions.Connected. New trust is 0.000004.","metadata":{"peer":"91.192.62.143","pid":7}}
{"timestamp":"2019-08-07 20:29:28.648721Z","level":"Debug","source":{"module":"Trust_system__Peer_trust","location":"File \"lib/trust_system/peer_trust.ml\", line 120, characters 56-63"},"message":"Decreasing trust for peer $peer due to action Trust_system.Actions.Sent_old_gossip. New trust is 0.000004.","metadata":{"peer":"91.192.62.143","pid":7}}
{"timestamp":"2019-08-07 20:29:28.648898Z","level":"Debug","source":{"module":"Network_pool__Network_pool_base","location":"File \"lib/network_pool/network_pool_base.ml\", line 32, characters 60-67"},"message":"Pool diff apply feedback: no useful transactions","metadata":{"pid":7}}
{"timestamp":"2019-08-07 20:29:30.315905Z","level":"Debug","source":{"module":"Trust_system__Peer_trust","location":"File \"lib/trust_system/peer_trust.ml\", line 120, characters 56-63"},"message":"Decreasing trust for peer $peer due to action Trust_system.Actions.Connected. New trust is -0.000020.","metadata":{"peer":"185.159.129.189","pid":7}}
{"timestamp":"2019-08-07 20:29:30.558054Z","level":"Debug","source":{"module":"Network_pool__Network_pool_base","location":"File \"lib/network_pool/network_pool_base.ml\", line 32, characters 60-67"},"message":"Pool diff apply feedback: Worse fee or already in pool","metadata":{"pid":7}}
{"timestamp":"2019-08-07 20:29:30.573332Z","level":"Info","source":{"module":"Gossip_net","location":"File \"lib/gossip_net/gossip_net.ml\", line 281, characters 65-72"},"message":"On RPC call, reconnected to a disconnected peer: ((host 185.159.129.189) (discovery_port 8303) (communication_port 8302))","metadata":{"pid":7}}
{"timestamp":"2019-08-07 20:29:30.573598Z","level":"Info","source":{"module":"Gossip_net","location":"File \"lib/gossip_net/gossip_net.ml\", line 210, characters 55-62"},"message":"Clearing disconnected peer set : (((host 185.159.129.189) (discovery_port 8303) (communication_port 8302)))","metadata":{"pid":7}}
{"timestamp":"2019-08-07 20:29:30.573830Z","level":"Info","source":{"module":"Gossip_net","location":"File \"lib/gossip_net/gossip_net.ml\", line 219, characters 55-62"},"message":"Restarting Kademlia","metadata":{"pid":7}}
{"timestamp":"2019-08-07 20:29:30.577315Z","level":"Debug","source":{"module":"Coda_run","location":"File \"app/cli/src/coda_run.ml\", line 70, characters 52-59"},"message":"Successfully wrote the visualization of the registered masks at location: /root/.coda-config/registered_masks.dot.","metadata":{"coda_run":"Logging state before program ends","pid":7}}
{"timestamp":"2019-08-07 20:29:30.693629Z","level":"Debug","source":{"module":"Coda_run","location":"File \"app/cli/src/coda_run.ml\", line 80, characters 58-65"},"message":"Successfully wrote the visualization of the transition frontier at location: /root/.coda-config/frontier.dot.","metadata":{"coda_run":"Logging state before program ends","pid":7}}
```

### PR DESCRIPTION
**Pull request by enolan**
I fixed a bunch of bugs related to killing child processes, concurrency, fds etc recently and a lot of that code was duplicated so I put the logic into a nice library. There shouldn't be any observable changes except in the logging.

Checklist:

- [x] Tests were added for the new behavior
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [x] Does this close issues? List them:

Closes #3128

### CODE REVIEW

File: src/lib/child_processes/child_processes.ml [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+          let%bind () = assert_lock_does_not_exist () in
+          let exit_or_signal = Or_error.ok_exn kill_res in
+          [%test_eq: Unix.Exit_or_signal.t] exit_or_signal
+            (Error (`Signal Signal.term)) ;
+          assert (Option.is_some @@ termination_status process) ;
+          Deferred.unit )
+
+    let () =
+      Async.Scheduler.set_record_backtraces true ;
+      Core.Backtrace.elide := false
```
Comment (cmr) [inline src/lib/child_processes/child_processes.ml]: remove these or put them at the top of the module imo?

File: src/lib/child_processes/child_processes.ml [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+    (Strict_pipe.Reader.iter_without_pushback logging_r ~f:(fun line ->
+         match log with
+         | `Log level -> (
+             let simple_log_msg =
+               lazy
+                 { Logger.Message.timestamp= Time.now ()
+                 ; level
+                 ; source=
+                     Logger.Source.create ~module_:__MODULE__ ~location:__LOC__
+                 ; message= sprintf "Output from process %s: %s" name line
```
[the remark this answers is not in the document]
Comment (enolan) [reply src/lib/child_processes/child_processes.ml]: This function is already passed a logger extended with "process" and "handle" metadata fields. Do you think the PID is useful in addition to that?

File: src/lib/child_processes/child_processes.ml (hunk 1 above)
Comment (enolan) [reply src/lib/child_processes/child_processes.ml]: yeah, done

File: src/lib/child_processes/child_processes.mli [hunk 3]
```diff
@@ -0,0 +1,59 @@
+(** Management of starting, tracking, and killing child processes. *)
+
+open Async
+open Pipe_lib
+
+type t
```
Comment (psteckler) [inline src/lib/child_processes/child_processes.mli]: (* a child process *)

File: src/lib/child_processes/child_processes.mli [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+exception Child_died
+
+(** A pipe of the standard out of the process, grouped by line. *)
+val stdout_lines : t -> string Strict_pipe.Reader.t
+
+(** Same for standard error. *)
+val stderr_lines : t -> string Strict_pipe.Reader.t
+
+(** Writer to process's stdin *)
```
Comment (psteckler) [inline src/lib/child_processes/child_processes.mli]: `process'`

File: src/lib/child_processes/child_processes.mli [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+val stdin : t -> Writer.t
+
+(** [None] if the process is still running, [Some] when it's exited *)
+val termination_status : t -> Unix.Exit_or_signal.t option
+
+(** What to do with one of standard out or standard error. If the first
+    component is [`Log] and the process outputs valid messages in our JSON
+    format they will be passed through unmodified regardless of the level.
+    Otherwise they'll be wrapped and logged at the specified level.
+*)
```
Comment (psteckler) [inline src/lib/child_processes/child_processes.mli]: Sorry, what do `Pipe`, `No_pipe` do?

File: src/lib/child_processes/child_processes.mli [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+type output_handling =
+  [`Log of Logger.Level.t | `Don't_log] * [`Pipe | `No_pipe]
+
+(** Start a process, handling a lock file, termination, optional logging, and
+    the standard in, out and error fds. *)
+val start_custom :
```
Comment (psteckler) [inline src/lib/child_processes/child_processes.mli]: why `_custom` ?

File: src/lib/child_processes/child_processes.mli [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+     logger:Logger.t
+  -> name:string
+     (** The name of the executable file, without any coda- prefix *)
+  -> checkout_relative_path:string
```
Comment (psteckler) [inline src/lib/child_processes/child_processes.mli]: `git_root_relative_path` maybe ?

File: src/lib/file_system/file_system.ml [hunk 8]
```diff
@@ -6,9 +7,14 @@ let remove_dir dir =
   let%bind _ = Process.run_exn ~prog:"rm" ~args:["-rf"; dir] () in
   Deferred.unit
 
-let try_finally ~(f : unit -> unit Deferred.t) ~finally =
-  let open Deferred.Or_error in
-  try_with f >>= (fun () -> ok_unit) |> ok_exn |> Deferred.bind ~f:finally
+let try_finally ~(f : unit -> 'a Deferred.t)
```
Comment (psteckler) [inline src/lib/file_system/file_system.ml]: this seems of general purpose, so maybe should be in a library other than `File_system`

File: src/lib/coda_net2/dune [hunk 9]
```diff
@@ -1,6 +1,6 @@
 (library
  (name coda_net2)
  (public_name coda_net2)
- (libraries core async network_peer file_system yojson coda_digestif pipe_lib envelope logger base58)
+ (libraries async base58 child_processes coda_digestif core envelope file_system logger network_peer pipe_lib yojson)
```
Comment (psteckler) [inline src/lib/coda_net2/dune]: nice alphabetization

File: src/lib/child_processes/tester.sh [hunk 10]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+
```
Comment (psteckler) [inline src/lib/child_processes/tester.sh]: Should have a comment indicating how this script is to be used.

File: src/lib/child_processes/child_processes.ml [hunk 11]
```diff
@@ -0,0 +1,444 @@
...
+            return (Ok r)
+        | Error e ->
+            go (Error e) xs )
+  in
+  go (Or_error.error_string "empty input") xs
+
+(* Unfortunately, `dune runtest` runs in a pwd deep inside the build directory.
+   This hack finds the project root by recursively looking for the dune-project
+   file. *)
+let get_project_root () =
```
Comment (psteckler) [inline src/lib/child_processes/child_processes.ml]: `get_git_root`

You don't want to run `git` for this: `git rev-parse --show-toplevel`

File: src/lib/child_processes/child_processes.ml [hunk 12]
```diff
@@ -0,0 +1,444 @@
...
+       deference the symlink if possible. *)
+    Deferred.Or_error.return
+      (Unix.getpid () |> Pid.to_int |> sprintf "/proc/%d/exe")
+
+(* Check the PID file, and if it exists and corresponds to a currently running
+   process, kill that process. This runs when the daemon starts, and should
+   *not* be used to kill a process that was started during this run of the
+   daemon.
+*)
+let maybe_kill_and_unlock : string -> string -> Logger.t -> unit Deferred.t =
```
Comment (psteckler) [inline src/lib/child_processes/child_processes.ml]: can `lockpath` be of type `Core_kernel.Filename.t` (which is just a string, but makes the role of the argument clear)

File: src/lib/file_system/file_system.ml (hunk 8 above)
Comment (enolan) [reply src/lib/file_system/file_system.ml]: I agree, but there's no obvious place to put it, and it was in `File_system` already.

File: src/lib/child_processes/child_processes.ml (hunk 11 above)
Comment (enolan) [reply src/lib/child_processes/child_processes.ml]: You could be running the tests from an sdist

File: src/lib/child_processes/tester.sh (hunk 10 above)
Comment (enolan) [reply src/lib/child_processes/tester.sh]: :heavy_check_mark:

File: src/lib/child_processes/child_processes.mli (hunk 7 above)
Comment (enolan) [reply src/lib/child_processes/child_processes.mli]: done

File: src/lib/child_processes/child_processes.mli (hunk 3 above)
Comment (enolan) [reply src/lib/child_processes/child_processes.mli]: :heavy_check_mark:

File: src/lib/child_processes/child_processes.mli (hunk 6 above)
Comment (enolan) [reply src/lib/child_processes/child_processes.mli]: added to the doc comment

File: src/lib/child_processes/child_processes.ml (hunk 12 above)
Comment (enolan) [reply src/lib/child_processes/child_processes.ml]: :heavy_check_mark:

File: src/lib/child_processes/child_processes.mli (hunk 5 above)
Comment (enolan) [reply src/lib/child_processes/child_processes.mli]: improved comment

**Comment by enolan** [comment]
@psteckler addressed your feedback

**Pull request merged**